drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -26,9 +21,11 @@ static int pd_intel_adsp_set_power_enable(struct pg_registers *reg, bool power_e
 	uint32_t SPA_bit_mask = BIT(reg->SPA_bit);
 
 	if (power_enable) {
-		sys_write32(sys_read32(reg->wr_address) | SPA_bit_mask, reg->wr_address);
+		sys_write32(sys_read32(ACE_DfPMCCU->dfpwrsts) | SPA_bit_mask,
```

Comment:
Should be `dfpwrctl` (as in control register). `Sts` is for reading status.

Reply:
This should work since `sts` value should match `ctl` value, but it's not the same behavior as before. 
sys_write32(sys_read32(ACE_DfPMCCU->**dfpwrctl**) | SPA_bit_mask, ACE_DfPMCCU->**dfpwrctl**);